pkg/apiserver/rest/utils/cache.go

```diff
@@ -16,26 +16,88 @@ limitations under the License.
 
 package utils
 
-import "time"
+import (
+	"context"
+	"sync"
+	"time"
+)
 
-// MemoryCache memory cache, support time expired
-type MemoryCache struct {
+// memoryCache memory cache, support time expired
+type memoryCache struct {
 	data          interface{}
 	cacheDuration time.Duration
 	startTime     time.Time
 }
 
 // NewMemoryCache new memory cache instance
-func NewMemoryCache(data interface{}, cacheDuration time.Duration) *MemoryCache {
-	return &MemoryCache{data: data, cacheDuration: cacheDuration, startTime: time.Now()}
+func newMemoryCache(data interface{}, cacheDuration time.Duration) *memoryCache {
+	mc := &memoryCache{data: data, cacheDuration: cacheDuration, startTime: time.Now()}
+	return mc
 }
 
 // IsExpired whether the cache data expires
-func (m *MemoryCache) IsExpired() bool {
+func (m *memoryCache) IsExpired() bool {
+	if m.cacheDuration == 0 {
+		return false
+	}
 	return time.Now().After(m.startTime.Add(m.cacheDuration))
 }
 
 // GetData get cache data
-func (m *MemoryCache) GetData() interface{} {
+func (m *memoryCache) GetData() interface{} {
 	return m.data
 }
+
+// MemoryCacheStore a sample memory cache instance, if data set cache duration, will auto clear after timeout.
+type MemoryCacheStore struct {
+	store sync.Map
+	ctx   context.Context
+}
```

Comment:
Do we truly need store the ctx inside a struct? This isn't a recommended way. https://pkg.go.dev/context